eclair-core/src/main/scala/fr/acinq/eclair/router/SyncHandlers.scala

```diff
@@ -0,0 +1,203 @@
+    s.to ! filter
+
+    // clean our sync state for this peer: we receive a SendChannelQuery just when we connect/reconnect to a peer and
+    // will start a new complete sync process
+    d.copy(sync = d.sync - s.remoteNodeId)
+  }
+
+  def handleQueryChannelRange(channels: SortedMap[ShortChannelId, PublicChannel], routerConf: RouterConf, origin: RemoteGossip, q: QueryChannelRange)(implicit ctx: ActorContext, log: LoggingAdapter): Unit = {
+    implicit val sender: ActorRef = ctx.self // necessary to preserve origin when sending messages to other actors
+    ctx.sender ! TransportHandler.ReadAck(q)
+    Kamon.runWithContextEntry(remoteNodeIdKey, origin.nodeId.toString) {
+      Kamon.runWithSpan(Kamon.spanBuilder("query-channel-range").start(), finishSpan = true) {
+        log.info("received query_channel_range with firstBlockNum={} numberOfBlocks={} extendedQueryFlags_opt={}", q.firstBlockNum, q.numberOfBlocks, q.tlvStream)
+        // keep channel ids that are in [firstBlockNum, firstBlockNum + numberOfBlocks]
+        val shortChannelIds: SortedSet[ShortChannelId] = channels.keySet.filter(keep(q.firstBlockNum, q.numberOfBlocks, _))
+        log.info("replying with {} items for range=({}, {})", shortChannelIds.size, q.firstBlockNum, q.numberOfBlocks)
+        val chunks = Kamon.runWithSpan(Kamon.spanBuilder("split-channel-ids").start(), finishSpan = true) {
+          split(shortChannelIds, q.firstBlockNum, q.numberOfBlocks, routerConf.channelRangeChunkSize)
+        }
+
+        Kamon.runWithSpan(Kamon.spanBuilder("compute-timestamps-checksums").start(), finishSpan = true) {
+          chunks.foreach { chunk =>
+            val reply = Router.buildReplyChannelRange(chunk, q.chainHash, routerConf.encodingType, q.queryFlags_opt, channels)
+            origin.peerConnection ! reply
+          }
+        }
+      }
+    }
+  }
+
+  def handleReplyChannelRange(d: Data, routerConf: RouterConf, origin: RemoteGossip, r: ReplyChannelRange)(implicit ctx: ActorContext, log: LoggingAdapter): Data = {
+    implicit val sender: ActorRef = ctx.self // necessary to preserve origin when sending messages to other actors
+    ctx.sender ! TransportHandler.ReadAck(r)
+
+    Kamon.runWithContextEntry(remoteNodeIdKey, origin.nodeId.toString) {
+      Kamon.runWithSpan(Kamon.spanBuilder("reply-channel-range").start(), finishSpan = true) {
+
+        @tailrec
+        def loop(ids: List[ShortChannelId], timestamps: List[ReplyChannelRangeTlv.Timestamps], checksums: List[ReplyChannelRangeTlv.Checksums], acc: List[ShortChannelIdAndFlag] = List.empty[ShortChannelIdAndFlag]): List[ShortChannelIdAndFlag] = {
+          ids match {
+            case Nil => acc.reverse
+            case head :: tail =>
+              val flag = computeFlag(d.channels)(head, timestamps.headOption, checksums.headOption, routerConf.requestNodeAnnouncements)
```

Comment:
We should move `computeFlag` inside this file as well.

Reply:
Done with dc79e8d.